src/theory/rewriter.cpp

```diff
@@ -210,16 +210,31 @@ Node Rewriter::rewriteTo(theory::TheoryId theoryId,
         // Rewrite until fix-point is reached
         for(;;) {
           // Perform the pre-rewrite
+          Kind originalKind = rewriteStackTop.d_node.getKind();
           RewriteResponse response = preRewrite(
               rewriteStackTop.getTheoryId(), rewriteStackTop.d_node, tcpg);
 
           // Put the rewritten node to the top of the stack
           rewriteStackTop.d_node = response.d_node;
           TheoryId newTheory = theoryOf(rewriteStackTop.d_node);
-          // In the pre-rewrite, if changing theories, we just call the other theories pre-rewrite
+          // In the pre-rewrite, if changing theories, we just call the other
+          // theories pre-rewrite. If the kind of the node was changed, then we
+          // pre-rewrite again.
           if (newTheory == rewriteStackTop.getTheoryId()
+              && originalKind == rewriteStackTop.d_node.getKind()
```

Comment:
Is it the case that we can remove the `theoryOf` condition here?  Perhaps it is implied by the kind not changing now?

Reply:
Yes, you are right. I am going to remove that condition.